datadog/data_source_datadog_synthetics_test_.go

```diff
@@ -0,0 +1,81 @@
+package datadog
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDatadogSyntheticsTest() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this data source to retrieve a Datadog Synthetic Test.",
+		ReadContext: dataSourceDatadogSyntheticsTestRead,
+
+		Schema: map[string]*schema.Schema{
+			"test_id": {
+				Description: "The synthetic test id to search for",
+				Type:        schema.TypeString,
+				Required:    true,
+				// ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-z]{3}-[a-z]{3}-[a-z]{3}$`), "must be of format abc-def-ghi"),
+			},
+			"name": {
+				Description: "The name of the synthetic test.",
+				Type:        schema.TypeString,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+			},
+			"tags": {
+				Description: "A list of tags assigned to the synthetic test.",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+			},
+			"url": {
+				Description: "The start URL of the synthetic test.",
+				Type:        schema.TypeString,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceDatadogSyntheticsTestRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV1 := providerConf.DatadogClientV1
+	authV1 := providerConf.AuthV1
+
+	tests, httpresp, err := datadogClientV1.SyntheticsApi.ListTests(authV1)
+	if err != nil {
+		return utils.TranslateClientErrorDiag(err, httpresp, "error getting synthetic tests")
+	}
+	if err := utils.CheckForUnparsed(tests); err != nil {
+		return diag.FromErr(err)
+	}
+
+	searchedId := d.Get("test_id").(string)
+	var matchedTests []datadog.SyntheticsTestDetails
+
+	for _, test := range *tests.Tests {
+		if test.GetPublicId() == searchedId {
+			matchedTests = append(matchedTests, test)
```

Comment:
Can we break here? We don't handle multiple results anyway, and we're testing an ID some presumably it's unique?